### ISSUE TITLE
run_cmd_qa does not capture full output

### ISSUE THREAD

**Issue by vanzod** [easybuilders/easybuild-framework#3593]
When WRF build fails as described in https://github.com/easybuilders/easybuild-easyconfigs/issues/12271 no output from the `configure` script is reported in the build log.

Here is the standard output obtained by executing the `configure` script manually:
```
checking for perl5... no
checking for perl... found /usr/bin/perl (perl)
Will use NETCDF in dir: /cvmfs/hc44rs.azure/centos/7.7/software/netCDF/4.7.4-gompi-2020a
Will use HDF5 in dir: /cvmfs/hc44rs.azure/centos/7.7/software/HDF5/1.10.6-gompi-2020a
Will use PHDF5 in dir: /cvmfs/hc44rs.azure/centos/7.7/software/HDF5/1.10.6-gompi-2020a

ERROR ERROR ERROR ERROR ERROR ERROR ERROR
'm4' utility not found! Can not configure.

If on an Ubuntu machine, use the command
         sudo apt-get install m4        
To download and install the 'm4' utility
```

while the standard error shows:
```
./configure: line 621: type: m4: not found
```

**Comment by boegel** [comment]
I'm looking into what's wrong with `run_cmd_qa` that it's not catching the output.

I've managed to reproduce the issue with a simple test script that basically does `echo hello` + `exit 1`, but I haven't figured a fix for the issue yet... `run_cmd_qa` isn't catching the output if the command completes very quickly, basically...

**Comment by akesandgren** [comment]
Incorrect flushing?

**Comment by boegel** [comment]
Problem found and fixed in #3599 \o/
